gen/encoder.go

```diff
@@ -195,7 +214,7 @@ func (g *Generator) notEmptyCheck(t reflect.Type, v string) string {
 	}
 
 	switch t.Kind() {
-	case reflect.Slice, reflect.Map:
+	case reflect.Slice, reflect.Map: // note: Array types don't have a useful empty value
```

Comment:
I put the comment there because everywhere else Array takes a similar code path to Slice.
Still want it moved?